### ISSUE TITLE
Loss issue

### ISSUE THREAD

**Issue by jaideep11061982** [meituan/YOLOv6#438]
I am continously getting an error below after epoch 1 or 2.
Model used yolov6n.
`!python  ./tools/train.py --batch 4 --img {IMG_SIZE} --epochs 5 --conf configs/yolov6n.py --data data/data.yaml --device 0
`
`     1/4    0.9659         0     188.1:  26%|██▌       | 311/1189 [07:00<19:46/usr/local/src/pytorch/aten/src/ATen/native/cuda/Loss.cu:111: operator(): block: [381,0,0], thread: [0,0,0] Assertion `input_val >= zero && input_val <= one` failed.`
```

11: operator(): block: [381,0,0], thread: [23,0,0] Assertion `input_val >= zero && input_val <= one` failed.
/usr/local/src/pytorch/aten/src/ATen/native/cuda/Loss.cu:111: operator(): block: [381,0,0], thread: [24,0,0] Assertion `input_val >= zero && input_val <= one` failed.
```

**Comment by shensheng272** [comment]
Did your use lasted code? And it's on your custom dataset?
Batch 4 is quit small, your might try smaller lr.

**Comment by leedewdew** [comment]
Seems nan happens. If you didn't change any code, then there may be some mistakes in your labels.

**Comment by jaideep11061982** [comment]
> Did your use lasted code? And it's on your custom dataset? It there more infos with the Assertion Batch 4 is quit small, you might try smaller lr.

Yes i did git clone, it was same code i was running with Yolov5 uralytics repo, thought of trying down with Yolov6  I just changed the repo only

**Comment by RobinBram** [comment]
I have gotten tha same issue but with batchsize 32 and it occurs on random epoch anywhere from second to 80th. Training v6n with custom data. I changed no hyperparameters.

**Comment by Chilicyy** [comment]
hi, @jaideep11061982  @RobinBram , if you want to train yolov6 models on  your custom dataset, we recommend using this config file (yolov6n_finetune.py) for stable training.

**Comment by RobinBram** [comment]
I have used the finetune config. I have also tried to remove all augmentation. The loss goes down in the beginning but the it can go up again and hover at another level. I have also tried lowering the learning rate.

**Comment by jaideep11061982** [comment]
I suspect there are issues with the momentum settings of sgd .
I tried running yolov7 I get expected results .

**Comment by Chilicyy** [comment]
@jaideep11061982 from the notebook https://www.kaggle.com/code/jaideepvalani/barrier-reef-yolov6-subseq-area?scriptVersionId=105230064 you share, it seems that you use the config file for training coco . We suggest to use  yolov6n_finetune.py for stable training.

**Comment by Chilicyy** [comment]
@RobinBram would you please share the training commands or some config params ?

**Comment by RobinBram** [comment]
This is the error message I receive. `/opt/conda/conda-bld/pytorch_1659484683044/work/aten/src/ATen/native/cuda/Loss.cu:118: operator(): block: [25,0,0], thread: [64,0,0] Assertion `input_val >= zero && input_val <= one` failed.
/opt/conda/conda-bld/pytorch_1659484683044/work/aten/src/ATen/native/cuda/Loss.cu:118: operator(): block: [25,0,0], thread: [65,0,0] Assertion `input_val >= zero && input_val <= one` failed.
/opt/conda/conda-bld/pytorch_1659484683044/work/aten/src/ATen/native/cuda/Loss.cu:118: operator(): block: [25,0,0], thread: [66,0,0] Assertion `input_val >= zero && input_val <= one` failed.`
Training with the stadard yolov6n_finetune but with the learningrate divided by 10. The loss went down this time but after  240 epochs, it diverged up this time and I got the error above. 

This is hte command I ran: `CUDA_VISIBLE_DEVICES=0 python tools/train.py --batch 4 --conf configs/yolov6n_finetune.py --data data/data.yaml --device 0 --img-size 1920  --device 0`

**Comment by trungpham2606** [comment]
Hi @Chilicyy, I met a very weird situation when training my custom dataset.
My custom dataset was correctly converted into yolo format (verified by running visualize_data.py file).
I turned off all augmentation methods.
From some very first epochs, the mAP is pretty good, but then drop to 0 (the cls_loss is not stable, sometimes is inf, my dataset has 1 class though).
Can you tell me where the real problem come from ?

**Comment by Chilicyy** [comment]
hi @trungpham2606 @RobinBram , if you use a small batchsize, it is better to use a smaller learning rate from empirical experiments. My email is 419895133@qq.com, please send me your dataset for further investigation if available. Thanks!
